src/Domain/Stager.php

```diff
@@ -84,12 +108,28 @@ private function validatePreconditions(): void
         }
     }
 
-    private function runCommand(): void
+    /**
+     * @throws \Symfony\Component\Process\Exception\RuntimeException
+     */
+    private function runCommand(): string
     {
-        $this->processFactory
+        $process = $this->processFactory
             ->create(array_merge([
                 'composer',
-                sprintf('--working-dir=%s', $this->stagingDir),
-            ], $this->command));
+                "--working-dir={$this->stagingDir}",
+            ], $this->composerCommand));
+        try {
+            $process->mustRun();
+            return $this->parseOutput($process);
+        } catch (\Symfony\Component\Process\Exception\ProcessFailedException $e) {
+            throw new ProcessFailedException($e->getMessage());
```

Comment:
If we're gonna wrap an exception in an exception, we should supply the previous exception as well for debugging purposes.

Reply:
Ooh, good thinking. I'll do that.